src/constraint_system/composer.rs

```diff
@@ -610,26 +618,72 @@ mod tests {
     }
 
     #[test]
-    fn test_conditional_select() {
-        let res = gadget_tester(
+    fn test_gadget() {
+        let mut t = PlookupTable4Arity::new();
+        t.insert_special_row(
+            BlsScalar::from(12),
+            BlsScalar::from(12),
+            BlsScalar::from(12),
+            BlsScalar::from(12),
+        );
+        t.insert_special_row(
+            BlsScalar::from(3),
+            BlsScalar::from(0),
+            BlsScalar::from(12),
+            BlsScalar::from(341),
+        );
+        t.insert_special_row(
+            BlsScalar::from(341),
+            BlsScalar::from(341),
+            BlsScalar::from(10),
+            BlsScalar::from(10),
+        );
+        let res = gadget_plookup_tester(
             |composer| {
-                let bit_1 = composer.add_input(BlsScalar::one());
-                let bit_0 = composer.add_input(BlsScalar::zero());
-
-                let choice_a = composer.add_input(BlsScalar::from(10u64));
-                let choice_b = composer.add_input(BlsScalar::from(20u64));
-
-                let choice = composer.conditional_select(bit_1, choice_a, choice_b);
-                composer.assert_equal(choice, choice_a);
-
-                let choice = composer.conditional_select(bit_0, choice_a, choice_b);
-                composer.assert_equal(choice, choice_b);
+                let twelve = composer.add_witness_to_circuit_description(BlsScalar::from(12));
+                let three = composer.add_witness_to_circuit_description(BlsScalar::from(3));
+                let ten = composer.add_witness_to_circuit_description(BlsScalar::from(10));
+                let zero = composer.add_witness_to_circuit_description(BlsScalar::from(0));
+                let three_four_one =
+                    composer.add_witness_to_circuit_description(BlsScalar::from(341));
+                composer.plookup_gate(twelve, twelve, twelve, Some(twelve), BlsScalar::zero());
+                composer.plookup_gate(twelve, twelve, twelve, Some(twelve), BlsScalar::zero());
+                composer.plookup_gate(three, zero, twelve, Some(three_four_one), BlsScalar::zero());
+                composer.plookup_gate(
+                    three_four_one,
+                    three_four_one,
+                    ten,
+                    Some(ten),
+                    BlsScalar::zero(),
+                );
             },
-            32,
+            65,
+            t,
         );
         assert!(res.is_ok());
     }
 
+    #[test]
+    #[should_panic]
+    fn test_gadget_fail() {
+        let mut t = PlookupTable4Arity::new();
+        t.insert_special_row(
+            BlsScalar::from(12),
+            BlsScalar::from(12),
+            BlsScalar::from(12),
+            BlsScalar::from(12),
+        );
+        let res = gadget_plookup_tester(
```

Comment:
Should this result not be checked?

Reply:
This is added with a `#[should_panic]` and assert!(res.is_err());
As the element is missing from the table